src/NUnitConsole/nunit-console.tests/TeamCityAssemblyResolverTests.cs

```diff
@@ -0,0 +1,162 @@
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System.IO;
+
+using NUnit.ConsoleRunner.Utilities;
+using NUnit.Framework;
+
+namespace NUnit.ConsoleRunner.Tests
+{
+    [TestFixture]
+    public class TeamCityAssemblyResolverTests
+    {
+        [Test]
+        [TestCase("123-dwdwd", true, "123")]
+        [TestCase("a23-dwdwd", true, "a23")]
+        [TestCase(null, false, null)]
+        [TestCase("", false, null)]
+        [TestCase("  ", false, null)]
+        [TestCase("-32423df", false, null)]
+        [TestCase("323-", false, null)]
+        [TestCase("-", false, null)]
+        [TestCase(" - ", false, null)]
+        public void ShouldResolveFlowId(string id, bool expectedResolved, string expectedFlowId)
+        {
+            // Given
+
+            // When
+            string actualFlowId;
+            var actualResolved = TeamCityAssemblyResolver.TryGetFlowId(id, out actualFlowId);
+
+            // Then
+            Assert.AreEqual(expectedResolved, actualResolved);
+            Assert.AreEqual(expectedFlowId, actualFlowId);
+        }
+
+        [Test]
+        [TestCase(@"c:\foo.dll", true, "foo.dll")]
+        [TestCase(@"aa\bb\foo.dll", true, "foo.dll")]
+        [TestCase(null, false, null)]
+        [TestCase("", false, null)]
+        [TestCase("bb", false, null)]
+        [TestCase("foo.dll", false, null)]
+        public void ShouldResolveAssemblyName(string fullName, bool expectedResolved, string expectedAssemblyName)
+        {
+            // Given
+            if (fullName != null)
+            {
+                fullName = fullName.Replace('\\', Path.DirectorySeparatorChar);
+            }
+
+            // When
+            string actualAssemblyName;
+            var actualResolved = TeamCityAssemblyResolver.TryGetAssemblyName(fullName, out actualAssemblyName);
+
+            // Then
+            Assert.AreEqual(expectedResolved, actualResolved);
+            Assert.AreEqual(expectedAssemblyName, actualAssemblyName);
+        }
+
+        [Test]
+        public void ShouldResolveAssemblyWhenAssemblySuteRegistered()
```

Comment:
Spelling mistake in Suite and this test failed on Travis in .NET 2.0 with the following, can you investigate?

1) Failed : NUnit.ConsoleRunner.Tests.TeamCityAssemblyResolverTests.ShouldResolveAssemblyWhenAssemblySuteRegistered
		  Expected: True
		  But was:  False
		at NUnit.ConsoleRunner.Tests.TeamCityAssemblyResolverTests.ShouldResolveAssemblyWhenAssemblySuteRegistered () [0x00000] in <filename unknown>:0